app/ide-desktop/lib/dashboard/src/modals/SetOrganizationNameModal.tsx

```diff
@@ -0,0 +1,124 @@
+/**
+ * @file
+ *
+ * Modal for setting the organization name.
+ */
+import * as React from 'react'
+
+import * as reactQuery from '@tanstack/react-query'
+import clsx from 'clsx'
+import * as router from 'react-router'
+
+import * as authProvider from '#/providers/AuthProvider'
+import * as backendProvider from '#/providers/BackendProvider'
+import * as textProvider from '#/providers/TextProvider'
+
+import * as aria from '#/components/aria'
+import * as ariaComponents from '#/components/AriaComponents'
+
+import * as backendModule from '#/services/Backend'
+
+const PLANS_TO_SPECIFY_ORG_NAME = [backendModule.Plan.team, backendModule.Plan.enterprise]
+
+/**
+ * Modal for setting the organization name.
+ * Shows up when the user is on the team plan and the organization name is the default.
+ */
+export function SetOrganizationNameModal() {
+  const { getText } = textProvider.useText()
+
+  const { backend } = backendProvider.useBackend()
+  const { session } = authProvider.useAuth()
+
+  const userId = session && 'user' in session && session.user?.userId ? session.user.userId : null
+  const userPlan =
+    session && 'user' in session && session.user?.plan != null ? session.user.plan : null
+
+  const queryClient = reactQuery.useQueryClient()
+  const { data } = reactQuery.useSuspenseQuery({
+    queryKey: ['organization', userId],
+    queryFn: () => {
+      if (backend.type === backendModule.BackendType.remote) {
+        return backend.getOrganization().catch(() => null)
+      } else {
+        return null
+      }
+    },
+    staleTime: Infinity,
```

Comment:
probably a bad idea. default `staleTime` should be fine, if it's infinity there's a much higher risk of getting an old value (which can then be edited and submitted, losing any changes in the meantime)

Reply:
This dialog displays only when the Org is not specified. We can't remove the name of the organization after setting it up. So I think this shouldn't be a problem. Also, eventually, we will manually invalidate this query after changing the name of the organization.